docs/src/main/tut/typeclasses.md

```diff
@@ -0,0 +1,73 @@
+The typeclass pattern is a technique borrowed from haskell. You can
+think of it as an "interface" in the java sense, but universally
+quantified over some type. In scala they are encoded as a trait that
+has one type variable, such as:
```

Comment:
Err right. Arrow was a poor example. What about something like [Shapeless's Prepend](https://github.com/milessabin/shapeless/blob/7832d1b2429907548f32b04264215456b5009d63/core/src/main/scala/shapeless/ops/hlists.scala#L1491)? Is that not a true type class?